src/store/builder/index.ts

```diff
@@ -91,6 +95,26 @@ export const getters: GetterTree<IBuilderState, IRootState> = {
   exitLabelsById: (state, getters, { flow: { flows } }, rootGetters) => mapValues(keyBy(flatMap(flows[0].blocks, 'exits'), 'uuid'), 'label'),
 
   isEditable: (state) => state.isEditable,
+
+  selectedBlocks(_state, _getters, _rootState, rootGetters) {
+    return rootGetters['flow/selectedBlocks']
+  },
+
+  selectedBlockAtTheTopPosition(_state, getters) {
+    return minBy(getters.selectedBlocks, 'vendor_metadata.io_viamo.uiData.yPosition')
+  },
+
+  selectedBlockAtTheFurthestLeftPosition(_state, getters) {
+    return minBy(getters.selectedBlocks, 'vendor_metadata.io_viamo.uiData.xPosition')
+  },
+
+  isAnyLeftSpaceAvailable(_state, getters) {
+    return getters.selectedBlockAtTheFurthestLeftPosition.vendor_metadata.io_viamo.uiData.xPosition > 0
+  },
+
+  isAnyTopSpaceAvailable(state, getters) {
+    return getters.selectedBlockAtTheTopPosition.vendor_metadata.io_viamo.uiData.yPosition - state.toolbarHeight > 0
```

Comment:
@bzabos , [you comment about separating pixel](https://github.com/FLOIP/flow-builder/pull/101#discussion_r655485434) computation is still relevant here.
I introduced this `toolbarHeight` because we would like to know in JS side if there is no more space between the topBlock and the toolbar, and we know the toolbar has a dynamic height.

I tried you proposal 
> Using the same logic as we already use to show/hide alerts-toolbar, we can (create, then) apply .has-notification-alerts-banner class to canvas which can add a margin-top of whatever the magic number is. This mitigates mixing javascript and css, keeps pixel variations within css, and avoids having those magic numbers sprinkled throughout our formulas and vuex state.

... but I cannot resolve the challenge of removing this formula from JS. Any idea ?

<img width="628" alt="Screen Shot 2021-06-21 at 3 08 05 PM" src="https://user-images.githubusercontent.com/68745918/122832752-0f201980-d2a9-11eb-993f-660c10b0b944.png">